src/realm/impl/simulated_failure.cpp

```diff
@@ -168,7 +168,7 @@ bool SimulatedFailure::do_check_trigger(FailureType failure_type) noexcept
     return false;
 }
 
-#if REALM_ARCHITECTURE_X86_32 && REALM_IOS
+#if (REALM_ARCHITECTURE_X86_32 && REALM_IOS) || REALM_WATCHOS
```

Comment:
This is related to https://github.com/realm/realm-core/pull/5102. I am going to make a guess and say that we should probably include REALM_TVOS as well. Alternatively, we already have a `RLM_NO_THREAD_LOCAL` which we could move to features.h so it could be used here as well.

Reply:
The second option will help to avoid this to happen in other targets that we are not counting on, so will be a better solution to this issue. I can do it on this PR if we agree on this approach.

Reply:
I tested this approach but seems that it doesn't work for this particular scenario

Reply:
`RLM_NO_THREAD_LOCAL` is equal to `REALM_PLATFORM_APPLE`, which doesn't seem like a correct definition.

Reply:
Is thread-local storage actually not supported at all on watchOS? I would have expected it to be supported on 64-bit watchOS.